packages/create-cloudflare/src/frameworks/next/index.ts

```diff
@@ -72,16 +73,74 @@ const configure = async (ctx: PagesGeneratorContext) => {
 	writeFile(handlerPath, handlerFile);
 	updateStatus("Created an example API route handler");
 
+	const installEslintPlugin = await shouldInstallNextOnPagesEslintPlugin(ctx);
+
+	if (installEslintPlugin) {
+		await writeEslintrc(ctx);
+		updateStatus("eslint-plugin-next-on-pages added to eslintrc");
+	}
+
 	// Add some dev dependencies
 	process.chdir(projectName);
-	const packages = ["@cloudflare/next-on-pages@1", "vercel"];
+	const packages = [
+		"@cloudflare/next-on-pages@1",
+		"vercel",
+		...(installEslintPlugin ? ["eslint-plugin-next-on-pages"] : []),
+	];
 	await installPackages(packages, {
 		dev: true,
 		startText: "Adding the Cloudflare Pages adapter",
 		doneText: `${brandColor(`installed`)} ${dim(packages.join(", "))}`,
 	});
 };
 
+// Note: this isn't a generic helper since it's Next specific, this can be
+//       made into a generic helper by checking all possible types of configs
+//       (including a field in the package.json)
+//       (and potentially returning what type of config is used)
+export const usesEslint = (projectRoot = ".") => {
+	// TODO: check other types of config files, etc...
+	return existsSync(`${projectRoot}/.eslintrc.json`);
+};
+
+export const shouldInstallNextOnPagesEslintPlugin = async (
+	ctx: PagesGeneratorContext
+): Promise<boolean> => {
+	if (!usesEslint(ctx.project.name)) return false;
+
+	// TODO: ask james about this
+	// I want to ask the question but I think we should also allow users to
+	// provide this via a flag in case they don't want any interactivity at all
+	return await processArgument(ctx.args, "eslint-plugin" as keyof C3Args, {
+		type: "confirm",
+		question: "Do you want to use the next-on-pages eslint-plugin?",
+		label: "",
+		defaultValue: true,
+	});
+};
+
+export const writeEslintrc = async (
+	ctx: PagesGeneratorContext
+): Promise<void> => {
+	// TODO: ask james, is it ok to completely overriding the file?
+	//       I could parse it and modify it but there the problem that the
+	//       .eslintrc.json can also have comments so if they were to add
+	//       some this would fail (maybe it's fine as we're going to include this
+	//       in future (pages-)e2e tests?)
+	const eslintConfig = {
+		plugins: ["eslint-plugin-next-on-pages"],
+		extends: [
+			"next/core-web-vitals",
+			"plugin:eslint-plugin-next-on-pages/recommended",
+		],
+	};
+
+	writeFileSync(
+		`${ctx.project.name}/.eslintrc.json`,
+		JSON.stringify(eslintConfig, null, "  ") + "\n"
+	);
```

Comment:
is it ok to completely overriding the file? or should we update it instead?

I could parse it and modify it but there's the problem that the `.eslintrc.json` can also have comments so if the Next team were to add some this parsing would fail and need some complex rewriting in order to keep the comments untouched

maybe it would be fine for now to update the file without including potential future comments as we're going to include this in future (pages-)e2e tests? (so if something were to break in the future we'd notice right away and we'd be able to amend it)

Reply:
`.json` can have comments 😄?

Updating is better than overwriting in general since there's less chance of us clobbering important things put there by the framework. We've write/overwrite files in a few places where code-modding would be super hard or where there's a low chance an existing file will be there.

Reply:
> .json can have comments 😄?

unfortunately yes, some jsons do allow comments 😓 (like tsconfigs) and the comments are also (at least that was the case last time I tried) quite complex to work with because they aren't easily and neatly parsed 😓

let me know what you prefer here, I am fine with overriding or if preferred I can update it 🙂
(but if I update it I would ignore comments if that's ok, as that as I mentioned would make things much much more complex I think, anyways Next not seem to produce any (at least for now))

Reply:
ok I'll do a `JSON.parse` then if that's ok 🙂 

anyways sorry my wording was a bit off, it's not really ignoring the comments, but ignoring the possibility of having comments 😅, so if the Next team were to add comments later down the line this piece of code would break, but hopefully it's fine and we can just react to that when/if that happens? 🙂

Reply:
`ok I'll do a JSON.parse then if that's ok 🙂` -- go for it!

`but hopefully it's fine and we can just react to that when/if that happens` -- absolutely